### ISSUE TITLE
pytorch-lightning

### ISSUE THREAD

**Issue by jeff-regier** [YosefLab/scvi-tools#693]
@adamgayoso @romain-lopez @PierreBoyeau --

I've had a pretty good experience on another project with pytorch-lightning, so I thought I'd mention it here. It might work even better for scVI. See
https://pytorch-lightning.readthedocs.io/en/stable/new-project.html

I like how lightning puts each dataloader in the same class as the model that it trains. It's also nice that boilerplate code (e.g. the training loop) can be removed entirely. Might help with #654

**Comment by adamgayoso** [comment]
This certainly does look nice, thanks for sharing! I can see the main advantage as cleaning up the training code and making it easier to create new models. Though it's unclear to me right now how this helps users access the latent variables or DE testing after training, which seems like it will still involve the user providing data, a dataloader being created, and passing data through the model, which is basically what all the `Posterior` functions do now.

**Comment by adamgayoso** [comment]
Revisiting this, pytorch lightning could simplify our training code.
